### ISSUE TITLE
Settings as environment variables not working

### ISSUE THREAD

**Issue by DominikPinsel** [eclipse-dataspaceconnector/DataSpaceConnector#608]
Currently the  environment variables settings are searched using the default setting keys (e.g. edc.web.port).
I assume the setting key should be made uppercase and the dots should be replaced by underscore.

From `DefaultServiceExtensionContext` 

```java
    @Override
    public String getSetting(String key, String defaultValue) {
        String value;
        for (ConfigurationExtension extension : configurationExtensions) {
            value = extension.getSetting(key);
            if (value != null) {
                return value;
            }
        }
        value = System.getProperty(key);
        if (value != null) {
            return value;
        }
        value = System.getenv(key); /* e.g. key="edc.web.port" is not a environment variable */
        return value != null ? value : defaultValue;
    }
```

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

**Comment by ndr-brt** [comment]
Good catch. Actually you posted an old piece of code, now the settings are obtained from a `Config` instance. Fixing this is a matter to map the environment variable keys to the "java properties" format before creating the config object here:
https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/32936a4eb6ecba5eb33617c75ea2cd119897757a/core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/DefaultServiceExtensionContext.java#L246

**Comment by DErCouch** [comment]
Suggest to mark system properties and system environment variables like the following, cause they are different in their meaning

1. System Property -> lowercase separated by dot (e.g., os.name)
2. System Environment -> uppercase separated by underscore (e.g., JAVA_HOME) 

Therefore in my opinion the better place to address this is directly in the ConfigFactory:

https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/32936a4eb6ecba5eb33617c75ea2cd119897757a/core/base/src/main/java/org/eclipse/dataspaceconnector/core/config/ConfigFactory.java#L31-L41

**Comment by paullatzelsperger** [comment]
> Good catch. Actually you posted an old piece of code, now the settings are obtained from a `Config` instance. Fixing this is a matter to map the environment variable keys to the "java properties" format before creating the config object here:
> 
> https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/32936a4eb6ecba5eb33617c75ea2cd119897757a/core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/DefaultServiceExtensionContext.java#L246

this is the way.
